sdcm/sla/libs/sla_utils.py

```diff
@@ -375,3 +385,31 @@ def clean_auth(entities_list_of_dict):
                         auth.drop()
                     except Exception as error:  # pylint: disable=broad-except
                         LOGGER.error("Failed to drop '%s'. Error: %s", auth.name, error)
+
+    @staticmethod
+    @retrying(n=50, sleep_time=3, message="Wait for service level has been propagated to all nodes",
+              allowed_exceptions=(Exception, SchedulerGroupNotFound, WrongServiceLevelShares,))
+    def wait_for_service_level_propagated(cluster, service_level, start_time, prometheus_stats):
```

Comment:
@soyacz 
Thanks, good to know. What is advantage when get metrics directly?
The method `.get_scylla_scheduler_shares_per_sla` is in use a while and tested, works OK. I do not see a reason to add new one

Reply:
In this case, it's even a necessity to poll from the node and not from prometheus for 2 reasons:
1. Prometheus itself has a polling interval, so we depended on this polling interval as well and it may mislead us.
2. If Prometheus failed to poll the information from one node from example since it's too loaded, it may wait another poll interval for this node, while our retry is faster.

Reply:
I would suggest doing it on a follow up PR, and see how stable it it would be with doing it per node directly.

Reply:
@fruch this change already performed. The test passed in this weekend. 
I am running another test after fixes by @soyacz comments